internal/services/tem/webhook.go

```diff
@@ -0,0 +1,193 @@
+package tem
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	tem "github.com/scaleway/scaleway-sdk-go/api/tem/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/httperrors"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/regional"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/services/account"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/types"
+)
+
+func ResourceWebhook() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ResourceWebhookCreate,
+		ReadContext:   ResourceWebhookRead,
+		UpdateContext: ResourceWebhookUpdate,
+		DeleteContext: ResourceWebhookDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"domain_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The domain id",
+			},
+			"region":     regional.Schema(),
+			"project_id": account.ProjectIDSchema(),
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.StringLenBetween(3, 127),
+			},
+			"event_types": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringNotInSlice([]string{"unknown_type"}, false),
+				},
+				Description: "List of event types",
+				MinItems:    1,
+			},
+			"sns_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "SNS ARN",
+				ValidateFunc: validation.StringLenBetween(3, 127),
+			},
+			"organization_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The organization id",
+			},
+			"id": {
```

Comment:
When is this Id updated? Why you put an id in the schema explicitely while it is not the case in other resources?